### PR TITLE
:bookmark: learning from scratch

### PR DESCRIPTION
**Pull request by Steve-YJ**
review deep learning learning method
SGD optimization
need to learn more optimizations like weight decay, dropout, batch-normalization

### CODE REVIEW

**Comment by Steve-YJ** [comment]
need to learn more and summarize it!
